examples/seismic/model.py

```diff
@@ -399,43 +399,8 @@ def initialize_function(function, data, nbpml, pad_mode='edge'):
     :param pad_mode: A string or a suitable padding function as explained in
                      :func:`numpy.pad`.
     """
-    pad_widths = [(nbpml, nbpml) for i in range(function.ndim)]
+    pad_widths = [(nbpml + i.left, nbpml + i.right) for i in function._offset_domain]
     data = np.pad(data, pad_widths, pad_mode)
-
-    distributor = function.grid.distributor
-
-    glb_ranges = distributor.glb_ranges
-    glb_shape = distributor.glb_shape
-
-    assert data.shape == glb_shape
-
-    # TODO: If running with MPI, all ranks have at this point built the entire
-    # global data; below, the local grid is extracted. However, this is not
-    # how things should work in practice -- the global function should have been
-    # memory-mapped at the very beginning, so that we avoid most of the following
-    # costly (in a real-life setting, prohibitively expensive) operations, but
-    # this still needs to be implemented
-    pad_widths = []
-    data_slices = []
-    for d, o in zip(function.dimensions, function._offset_domain):
-        try:
-            glb_range = glb_ranges[d]
-
-            lslice = min(glb_range.start, max(glb_range.start - o.left, 0))
-            rslice = max(glb_range.stop, min(glb_range.stop + o.right, glb_shape[d]))
-
-            lpad = o.left - glb_range.start if lslice == 0 else 0
-            rpad = o.right - (rslice - glb_range.stop) if rslice == glb_shape[d] else 0
-
-            data_slices.append((lslice, rslice))
-            pad_widths.append((lpad, rpad))
-        except KeyError:
-            # Not a distributed dimension, so we make sure to get the entire array
-            data_slices.append((None, None))
-            pad_widths.append((0, 0))
-    data = data[[slice(*i) for i in data_slices]]
-    data = np.pad(data, pad_widths, pad_mode)
-
     function.data_with_halo[:] = data
```

Comment:
Why is this gone? Does it really do the same?

Reply:
yes. The objective of this PR was hiding MPI almost completely away from user code. This is a symptom that this is indeed happening